app/cdap/components/GridTable/components/MenuItemComponent/types.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+export interface IMenuItem {
+  label?: string;
+  supportedDataType?: string[];
+  value?: string;
+  options?: IMenuItem[];
+  title?: string;
+  action?: string;
+  dataType?: string[];
+  icon?: JSX.Element;
+  toolName?: string;
+  open?: boolean;
+}
+
+export interface IMenuItemComponentProps {
+  item: IMenuItem;
+  index: number;
+  onMenuClick: (e: React.MouseEvent<HTMLLIElement, MouseEvent>, item: IMenuItem) => void;
```

Comment:
unsure if this is what you want to be using here https://developer.mozilla.org/en-US/docs/Web/API/HTMLLIElement ?

Reply:
updated to Element